examples/feature_selection/plot_feature_selection_pipeline.py

```diff
@@ -1,40 +1,92 @@
 """
 ==================
-Pipeline Anova SVM
+Pipeline ANOVA SVM
 ==================
 
-Simple usage of Pipeline that runs successively a univariate
-feature selection with anova and then a SVM of the selected features.
+This example shows how a feature selection can be easily integrated within
+a machine learning pipeline.
 
-Using a sub-pipeline, the fitted coefficients can be mapped back into
-the original feature space.
+We also show that you can easily introspect part of the pipeline.
 """
-from sklearn import svm
+# %%
+print(__doc__)
+from sklearn import set_config
+set_config(display='diagram')
+
+# %%
+# We will start by generating a binary classification dataset. Subsequently, we
+# will divide the dataset into two subsets.
+
+# %%
 from sklearn.datasets import make_classification
-from sklearn.feature_selection import SelectKBest, f_classif
-from sklearn.pipeline import make_pipeline
 from sklearn.model_selection import train_test_split
-from sklearn.metrics import classification_report
 
-print(__doc__)
-
-# import some data to play with
 X, y = make_classification(
-    n_features=20, n_informative=3, n_redundant=0, n_classes=4,
+    n_features=20, n_informative=3, n_redundant=0, n_classes=2,
     n_clusters_per_class=2)
-
 X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)
 
-# ANOVA SVM-C
-# 1) anova filter, take 3 best ranked features
-anova_filter = SelectKBest(f_classif, k=3)
-# 2) svm
-clf = svm.LinearSVC()
+# %%
+# A common mistake done with feature selection is indeed to search a subset
+# of discriminative features on the full dataset instead of only using the
+# training set. The usage of scikit-learn :func:`~sklearn.pipeline.Pipeline`
+# prevents to make such mistake.
+#
+# Here, we will demonstrate how to build a pipeline where the first step will
+# be the feature selection.
+
+# %%
+from sklearn.feature_selection import SelectKBest, f_classif
+from sklearn.pipeline import make_pipeline
+from sklearn.svm import LinearSVC
 
+anova_filter = SelectKBest(f_classif, k=3)
+clf = LinearSVC()
 anova_svm = make_pipeline(anova_filter, clf)
+anova_svm
+
+# %%
+# We created a pipeline where a feature selection will be the first step of
+# the processing before to use its output in an SVM classifier.
```

Comment:
I think this sentence is not needed. Since the sentence above

>  Here, we will demonstrate how to build a pipeline where the first step will be the feature selection.

states that the feature selection is the first step. And the next part describes the data flow in detail.

(I tried rewriting it because the grammar is a little strange to me and it ended up sounding like the next sentence you already have)